### ISSUE TITLE
tapestat statistics not working with RHEL7.9

### ISSUE THREAD

**Issue by jamthomp** [sysstat/sysstat#291]
Writing to an IBM LTO7 tape.  Not seeing any read/write stats coming through.

Sample stat:
Tape:    r/s     w/s   kB_read/s   kB_wrtn/s %Rd %Wr %Oa    Rs/s    Ot/s
st0        0       0           0               0             0       0      91       0       4

Installed Packages
sysstat.x86_64                                                                                 10.1.5-19.el7  

All the files in /sys/class/scsi_tape/st0/stats  are 0 except for:
io_ns = 5058457945440
other_cnt = 26770

**Comment by sysstat** [comment]
So it seems this problem is not related to sysstat? If all the stat files have 0 inside then tapestat will logically display 0 values...

**Comment by jamthomp** [comment]
After more testing.  If I use dd, then tapestat and the stat files work correctly.  But if the system writes with the application,  in this case IBM's HPSS software, it doesn't work.  Any idea how an application would bypass whatever triggers the stats files to get incremented?

**Comment by doug-gilbert** [comment]
IBM's HPSS might be using the Linux sg driver, or possibly the bsg driver. They bypass the st driver.
I'm adding st driver like statistics (similar place, same names (e.g. /sys/class/scsi_generic/sg3/stats/rd_cnt) experimentally to the sg driver. Maybe 'systat' could comment on how easy it would be to clone tapestat to a new sgstat utility. The "rd" and "wr" counts/timings are based on data direction as the sg driver is not in the business of decoding command codes.

**Comment by sysstat** [comment]
Would it make sense to use the same tapestat command to display stats for sg drives?
Or maybe a symlink (sgstat) to tapestat which would make it display them instead of tapes' stats?
I have the feeling that the code will be pretty the same and I'm not particularly excited about duplicating it...

**Comment by doug-gilbert** [comment]
Yes, that would be fine. The sysfs attributes have the same names as st. The interpretation of "read" and "write" is a bit different: for sg "read" is data-in (i.e. user data _from_ device) and "write" is data-out while "other" is no user data (e.g. SCSI TEST UNIT READY command). Practically this shouldn't make a significant difference, for example fetching VPD pages via the INQUIRY command will bump the read_cnt and the read_byte_cnt.
```
/sys/class/scsi_generic/sg0/stats$ ls
in_flight  other_cnt      read_cnt  resid_cnt       write_cnt
io_ns      read_byte_cnt  read_ns   write_byte_cnt  write_ns
```
Should attributes read via sysfs have a trailing line feed?

**Comment by sysstat** [comment]
No, attributes should just have a trailing carriage return.
